### ISSUE TITLE
Are we allowed to use code from Stack Overflow wholesale

### ISSUE THREAD

**Issue by nighoggDatatype** [nus-cs2103-AY2021S2/forum#268]
When I look up the policy on reuse, I found this: [https://nus-cs2103-ay2021s2.github.io/website/admin/appendixB-policies.html#policy-on-reuse](url)

What stumped me is that it both asks that "The work comes from a source of 'good standing' (such as an established open source project)." and it gives provisions to "reuse code snippets found on the Internet e.g. from StackOverflow answers", which is not an established source.

In my case, I'm trying to use [https://stackoverflow.com/a/13564498/11358676](url) for code on Levenshtein distance, and I'm largely copying code wholesale. Can I use it as is, or should I instead use an established third party library like "Apache Commons Lang3"?

**Comment by damithc** [comment]
> What stumped me is that it both asks that "The work comes from a source of 'good standing' (such as an established open source project)." and it gives provisions to "reuse code snippets found on the Internet e.g. from StackOverflow answers", which is not an established source.

The former applies to using third-party libraries only. You can reuse small code snippets from anywhere (e.g., stackoverflow) as long as you give credit to the source. I'll rephrase it to make it clearer.
